### PR TITLE
Coding Standards - General Practices created

### PR DESCRIPTION
**Pull request by alinakazzaa**
Added General Programming Practices of Coding Standards topic

 - Introduction
 - Naming Conventions
 - Formatting and Layout
 - Developement Style

### CODE REVIEW

**Comment by alinakazzaa** [comment]
There is a lot of content here so I will use a diagram for the next section where it would be more suitable

**Comment by harrythomas1998** [comment]
Well laid out, but quite long. Maybe replace some of the text with diagrams.

File: README.md [hunk 1]
```diff
@@ -1 +1,69 @@
-# Assignment
+# Handbook for Best Coding Practices
+
+## Coding Standards
+
+### General Programming Practices
+
+It is important to realise that your programming choices will reflect the quality
+of your code and impact the productivity levels of your colleagues. Multidots - an agency specialising in web development for content publishers [coding standards guideline](https://www.multidots.com/importance-of-code-quality-and-coding-standard-in-software-development/) highlights the importance of code quality and coding standards for improving efficiency, minimising risk of project failure, reducing complexity and simplifying debugging processes.
```
Comment (chriskavanagh97) [inline README.md:6]: minimising the risk**

File: README.md [hunk 2]
```diff
@@ -1 +1,69 @@
...
+
+* Use meaningful variable names throughout your code
+* Maintain a consistent naming style for classes, functions and files
+* Names should be explanatory of the object or the functionality of the methods that carry them. 
+
+#### Formatting and Layout
+
+To work together as a team, it is important that each developer effectively formats code,  tools and ensures code is well-segmented and easy to read. 
+
+* Use auto-linters provided by the developement environemnt your team is using
```
Comment (chriskavanagh97) [inline README.md]: development**

Comment (chriskavanagh97) [reply README.md]: In your paragraph development is spelt incorrectly each time

File: README.md [hunk 3]
```diff
@@ -1 +1,69 @@
...
+
+**Refactoring**
+
+Overly complex structures and unnecessary code can be avoided by effective refactoring practices which should be done by all developers before code review. 
+
+*Consider*
+
+* Project structure that will shape the file structure choices
+* Single task functions to simplify code and encourage code reusability
+* Parameters and how to make best use without over-parametarising methods
```
Comment (chriskavanagh97) [inline README.md]: parameterising **

Comment (chriskavanagh97) [reply README.md]: the best**

**Review by chriskavanagh97 — COMMENTED** [review]
Overall very good I would add some pictures as in such a long file it Is easy for people to lose engagement. Pictures will add colour and provide more engagement

**Pull request closed without merging**